### ISSUE TITLE
Circular Dependency Using "@EnableCouchDbRepositories"

### ISSUE THREAD

**Issue by snackerphi** [Majlanky/couch-slacker#47]
I have a main gradle project that has the spring `SpringBootApplication` named `MyApplication`.
I have a sub project that the main project depends on which contains the couch-slacker impliementation for CouchDB.  The sub project just generates a jar that is added to the main application.  The sub project contains `CouchdbConfig` class which extends `CouchSlackerConfiguration` and a number of interfaces that extend `CrudRepository` which should be automatically setup as spring-data repositories.
If we add the `@EnableCouchDbRepositories` on the `MyApplication` class, we get the following (edited) circular error message and the application fails:
```

***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────�?
|  methodValidationPostProcessor    defined in class path resource [org/springframework/boot/autoconfigure/validation/ValidationAutoConfiguration.class]
↑     ↓
|  couchdbClassARepository          defined in my.db.CouchdbClassARepository defined in @EnableCouchDbRepositories declared on MyApplication
↑     ↓
|  couchdbClassBRepository          defined in my.db.CouchdbClassBRepository defined in @EnableCouchDbRepositories declared on MyApplication
↑     ↓
|  couchdbClassCRepository          defined in my.db.CouchdbClassCRepository defined in @EnableCouchDbRepositories declared on MyApplication
↑     ↓
...   ...
↑     ↓
|  dbClient                         defined in class path resource [my.db/CouchdbConfig.class]
↑     ↓
|  com.groocraft.couchdb.slacker.CouchDbContext

```

I've tried to move the `@EnableCouchDbRepositories` to the `CouchdbConfig` class (because we don't want a dependency on the db implementation explicitly on the main project), but we get a similar error doing that as well.

I'm not sure how to fix this.  Does it matter that we have the couch-slacker in a sub project rather than in the main project?

**Comment by Majlanky** [comment]
Hello, can you provide me the project? I guess not but I am asking to be sure. If you are not able to provide me the whole project, can you show me all dependencies you are using in your project? I need to replicate the issue. I have few candidates that causes the issue already but need narrower scope... 

I think you are using `spring-boot-starter-validation` in your project right?

**Comment by snackerphi** [comment]
I can't give you this project, but I will try to create something that demonstrates the issue.  I don't see `spring-boot-starter-validation` explicitly being specified in our `build.gradle` file.  Running gradle dependencies doesn't show it either.

**Comment by snackerphi** [comment]
I've invited you to [cs-demo](https://github.com/snackerphi/cs-demo) project

**Comment by Majlanky** [comment]
Just want to inform you, I am working on it but have to say I do not understand what Spring is doing right now...

**Comment by snackerphi** [comment]
@Majlanky Thank you!  I'm glad to help if possible.

**Comment by Majlanky** [comment]
> @Majlanky Thank you! I'm glad to help if possible.

Sorry I deleted the last message. I just realize that the IDE and Gradle hides my changes fixes the issue thanks to caches 🤦 I just solved you problem in v2.3.1. Let me know if there is any other issue... :godmode:

**Comment by snackerphi** [comment]
Great! I'll try it out!
